--- a/tools/rkcommon.c	2022-03-27 07:51:16.211055136 +0300
+++ b/tools/rkcommon.c	2022-03-27 08:20:51.602224811 +0300
@@ -136,6 +136,7 @@
 	{ "rv1108", "RK11", 0x1800, false, RK_HEADER_V1 },
 	{ "rv1126", "110B", 0x10000 - 0x1000, false, RK_HEADER_V1 },
 	{ "rk1808", "RK18", 0x200000 - 0x2000, false, RK_HEADER_V1 },
+	{ "rk3566", "RK35", 0x10000 - 0x1000, false, RK_HEADER_V2 },
 	{ "rk3568", "RK35", 0x10000 - 0x1000, false, RK_HEADER_V2 },
 };
 
