diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
--- a/arch/arm64/boot/dts/rockchip/Makefile	2020-09-18 23:35:07.485856567 +0300
+++ b/arch/arm64/boot/dts/rockchip/Makefile	2020-10-01 22:11:05.361959455 +0300
@@ -38,4 +38,5 @@
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-rockpro64.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-sapphire-excavator.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399-station-p1.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3399pro-rock-pi-n10.dtb
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-station-p1.dts b/arch/arm64/boot/dts/rockchip/rk3399-station-p1.dts
--- /dev/null	2013-01-18 10:51:04.168000019 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-station-p1.dts	2021-01-01 20:35:28.161927261 +0200
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2017 T-Chip Intelligent Technology Co., Ltd
+ * Copyright (c) 2019 Markus Reichl <m.reichl@fivetechno.de>
+ */
+
+/dts-v1/;
+#include "rk3399-roc-pc-mezzanine.dts"
+
+/ {
+
+	model = "Firefly Station P1";
+	compatible = "firefly,station-p1", "firefly,roc-rk3399-pc-plus", "rockchip,rk3399";
+
+	aliases {
+		mmc0 = &sdmmc;
+		mmc1 = &sdhci; /* MMC boot device */
+	};
+
+	chosen {
+		bootargs = "swiotlb=1 spectre_v2=0 kpti=0 ssbd=force-off";
+	};
+
+&rk808{
+	rtc {
+		compatible = "rk808-rtc";
+		status = "disabled";
+	};
+};
+
+&i2c0 {
+	status = "okay";
+	hym8563: hym8563@51 {
+		compatible = "haoyu,hym8563";
+		reg = <0x51>;
+		interrupt-parent = <&gpio0>;
+		interrupts = <5 IRQ_TYPE_EDGE_FALLING>;
+		pinctrl-names = "default";
+		#clock-cells = <0>;
+		clock-frequency = <32768>;
+		clock-output-names = "xin32k";
+	};
+};
