--- a/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi	2022-04-23 14:34:53.819331574 +0300
+++ b/arch/arm64/boot/dts/rockchip/rk3399-nanopi4.dtsi	2022-04-23 13:06:39.495968459 +0300
@@ -23,10 +24,20 @@
 		mmc2 = &sdhci;
 	};
 
-	chosen {
+	chosen: chosen {
+		bootargs = "swiotlb=1 coherent_pool=1m";
 		stdout-path = "serial2:1500000n8";
 	};
 
+	mach: board {
+		compatible = "friendlyelec,board";
+		machine = "NANOPI4";
+		hwrev = <255>;
+		model = "NanoPi 4 Series";
+		nvmem-cells = <&cpu_id>;
+		nvmem-cell-names = "id";
+	};
+
 	clkin_gmac: external-gmac-clock {
 		compatible = "fixed-clock";
 		clock-frequency = <125000000>;
@@ -105,11 +116,11 @@
 		regulator-name = "vbus_typec";
 	};
 
-	gpio-keys {
+	keys: gpio-keys {
 		compatible = "gpio-keys";
 		autorepeat;
 		pinctrl-names = "default";
-		pinctrl-0 = <&power_key>;
+		pinctrl-0 = <&key_pins>;
 
 		power {
 			debounce-interval = <100>;
@@ -123,7 +134,7 @@
 	leds: gpio-leds {
 		compatible = "gpio-leds";
 		pinctrl-names = "default";
-		pinctrl-0 = <&status_led_pin>;
+		pinctrl-0 = <&leds_pin>;
 
 		status_led: led-0 {
 			gpios = <&gpio0 RK_PB5 GPIO_ACTIVE_HIGH>;
@@ -132,6 +143,36 @@
 		};
 	};
 
+	rockchip_suspend: rockchip-suspend {
+		compatible = "rockchip,pm-rk3399";
+		rockchip,sleep-debug-en = <1>;
+		rockchip,sleep-mode-config = <
+			(0
+			| RKPM_SLP_ARMPD
+			| RKPM_SLP_PERILPPD
+			| RKPM_SLP_DDR_RET
+			| RKPM_SLP_PLLPD
+			| RKPM_SLP_CENTER_PD
+			| RKPM_SLP_AP_PWROFF
+			)
+			>;
+		rockchip,wakeup-config = <
+			(0
+			| RKPM_GPIO_WKUP_EN
+			| RKPM_PWM_WKUP_EN
+			)
+			>;
+		rockchip,pwm-regulator-config = <
+			(0
+			| PWM2_REGULATOR_EN
+			)
+			>;
+		rockchip,power-ctrl =
+			<&gpio1 17 GPIO_ACTIVE_HIGH>,
+			<&gpio1 14 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&rk808 1>;
@@ -502,6 +543,10 @@
 	status = "okay";
 };
 
+/*&reboot_mode {
+	status = "okay";
+};*/
+
 &pcie_phy {
 	assigned-clock-parents = <&cru SCLK_PCIEPHY_REF100M>;
 	assigned-clock-rates = <100000000>;
@@ -524,7 +569,7 @@
 	};
 
 	gpio-leds {
-		status_led_pin: status-led-pin {
+		leds_pin: status-led-pin {
 			rockchip,pins = <0 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
@@ -554,7 +599,7 @@
 	};
 
 	rockchip-key {
-		power_key: power-key {
+		key_pins: key-pins {
 			rockchip,pins = <0 RK_PA5 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
@@ -608,6 +653,10 @@
 	status = "okay";
 };
 
+/*&rng {
+	status = "okay";
+};*/
+
 &saradc {
 	vref-supply = <&vcca1v8_s3>;
 	status = "okay";
@@ -615,6 +664,7 @@
 
 &sdhci {
 	bus-width = <8>;
+	disable-cqe;
 	mmc-hs200-1_8v;
 	non-removable;
 	status = "okay";
@@ -644,6 +694,7 @@
 	sd-uhs-sdr104;
 	vmmc-supply = <&vcc3v0_sd>;
 	vqmmc-supply = <&vcc_sdio>;
+	vqmmc-off-microvolt = <3000000>;
 	status = "okay";
 };
 
@@ -720,10 +771,14 @@
 };
 
 &usbdrd_dwc3_0 {
+	snps,xhci-slow-suspend-quirk;
+	snps,xhci-trb-ent-quirk;
 	status = "okay";
 };
 
 &usbdrd_dwc3_1 {
+	snps,xhci-slow-suspend-quirk;
+	snps,xhci-trb-ent-quirk;
 	dr_mode = "host";
 	status = "okay";
 };
