--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts	2019-12-20 20:57:00.246054818 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly.dts	2019-12-20 20:57:31.961896620 +0200
@@ -555,6 +555,7 @@
 &pcie0 {
 	ep-gpios = <&gpio4 RK_PD1 GPIO_ACTIVE_HIGH>;
 	num-lanes = <4>;
+	max-link-speed = <2>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie_clkreqn_cpm>;
 	status = "okay";
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2019-12-20 20:56:29.706206671 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4.dts	2019-12-20 20:53:54.082973045 +0200
@@ -485,7 +485,7 @@
 &pcie0 {
 	ep-gpios = <&gpio4 RK_PD3 GPIO_ACTIVE_HIGH>;
 	num-lanes = <4>;
-	max-link-speed = <1>;
+	max-link-speed = <2>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie_clkreqnb_cpm>;
 	vpcie12v-supply = <&vcc12v_dcin>;
--- a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2019-12-20 20:55:00.698646529 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2019-12-20 20:55:34.990477547 +0200
@@ -603,6 +603,7 @@
 &pcie0 {
 	ep-gpios = <&gpio2 RK_PD4 GPIO_ACTIVE_HIGH>;
 	num-lanes = <4>;
+	max-link-speed = <2>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pcie_perst>;
 	vpcie12v-supply = <&vcc12v_dcin>;
