--- a/drivers/net/wireless/rtl8192eu/os_dep/linux/ioctl_cfg80211.c	2022-08-19 23:15:47.284619286 +0300
+++ b/drivers/net/wireless/rtl8192eu/os_dep/linux/ioctl_cfg80211.c	2022-08-19 23:18:24.809314778 +0300
@@ -451,7 +451,11 @@
 	if (!rtw_cfg80211_allow_ch_switch_notify(adapter))
 		goto exit;
 
-	cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
+	#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
+		cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
+	#else
+		cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0);
+	#endif
 
 #else
 	int freq = rtw_ch2freq(ch);
@@ -5073,7 +5077,11 @@
 	return ret;
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
 static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev)
+#else
+static int cfg80211_rtw_stop_ap(struct wiphy *wiphy, struct net_device *ndev, unsigned int link_id)
+#endif
 {
 	_adapter *adapter = (_adapter *)rtw_netdev_priv(ndev);
 
@@ -10132,7 +10140,11 @@
 	rtw_cfg80211_indicate_scan_done(adapter, _TRUE);
 
 	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 11, 0)) || defined(COMPAT_KERNEL_RELEASE)
-	if (wdev->current_bss) {
+	#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 19, 2)
+		if (wdev->current_bss) {
+	#else
+		if (wdev->connected) {
+	#endif
 		RTW_INFO(FUNC_ADPT_FMT" clear current_bss by cfg80211_disconnected\n", FUNC_ADPT_ARG(adapter));
 		rtw_cfg80211_indicate_disconnect(adapter, 0, 1);
 	}
