--- a/drivers/net/wireless/hflps170/fullmac/bl_main.c	2022-05-29 05:16:22.046853126 +0300
+++ b/drivers/net/wireless/hflps170/fullmac/bl_main.c	2022-02-12 22:19:54.678249495 +0200
@@ -14587,7 +14587,7 @@
 +        memcpy(ndev->dev_addr, params->macaddr, ETH_ALEN);
 +    else {
 +        memcpy(ndev->dev_addr, bl_hw->wiphy->perm_addr, ETH_ALEN);
-+        ndev->dev_addr[5] ^= vif_idx;
++        ndev->dev_addr = ndev->dev_addr[5] ^ vif_idx;
 +    }
 +
 +    if (params) {
--- a/drivers/net/wireless/hflps170/fullmac/bl_defs.h	2022-05-30 09:10:20.360783516 +0300
+++ b/drivers/net/wireless/hflps170/fullmac/bl_defs.h	2022-05-30 09:10:40.404868961 +0300
@@ -37,6 +38,13 @@
 #define LEGACY_PS_ID   0
 #define UAPSD_ID       1
 
+/*
+ * A-PMDU buffer sizes
+ * According to IEEE802.11n spec size varies from 8K to 64K (in powers of 2)
+ */
+#ifndef IEEE80211_MAX_AMPDU_BUF
+#define IEEE80211_MAX_AMPDU_BUF IEEE80211_MAX_AMPDU_BUF_HT
+#endif
 
 /**
  * struct bl_bcn - Information of the beacon in used (AP mode)
