diff -Naur a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/efuse/rtw_efuse.c b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/efuse/rtw_efuse.c
--- a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/efuse/rtw_efuse.c	2021-09-21 17:36:33.310916696 +0000
+++ b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/efuse/rtw_efuse.c	2021-10-08 21:20:05.599131665 +0000
@@ -898,8 +898,9 @@
 	j = 0;
 
 	for (i = 0; i < mapLen; i++) {
-		if (i % 16 == 0)
+		if (i % 16 == 0) {
 			RTW_PRINT_SEL(RTW_DBGDUMP, "0x%03x: ", i);
+		}
 			_RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
 				, pEfuseHal->fakeEfuseInitMap[i]
 				, ((i + 1) % 16 == 0) ? "\n" : (((i + 1) % 8 == 0) ? "	  " : " ")
diff -Naur a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_mlme.c b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_mlme.c
--- a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_mlme.c	2021-09-21 17:36:33.350916604 +0000
+++ b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_mlme.c	2021-10-08 20:45:52.751078722 +0000
@@ -3134,12 +3134,14 @@
 	ssc_chk = rtw_sitesurvey_condition_check(padapter, _FALSE);
 	if( ssc_chk == SS_DENY_BUSY_TRAFFIC) {
 		#ifdef CONFIG_LAYER2_ROAMING
-		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE)
+		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE) {
 			RTW_INFO(FUNC_ADPT_FMT" need to roam, don't care BusyTraffic\n", FUNC_ADPT_ARG(padapter));
-		else
+		} else
 		#endif
+		{
 			RTW_INFO(FUNC_ADPT_FMT" exit BusyTraffic\n", FUNC_ADPT_ARG(padapter));
 			goto exit;
+		}
 	}
 	else if (ssc_chk != SS_ALLOW)
 		goto exit;
diff -Naur a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_recv.c b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_recv.c
--- a/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_recv.c	2021-09-21 17:36:33.378916540 +0000
+++ b/drivers/net/wireless/rockchip_wlan/rtl8822ce/core/rtw_recv.c	2021-10-08 21:06:53.364735113 +0000
@@ -3555,9 +3555,10 @@
 			int i;
 			RTW_INFO("############ type:0x%02x subtype:0x%02x #################\n", type, subtype);
 
-			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8)
+			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8) {
 				RTW_INFO("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X:\n", *(ptr + i),
 					*(ptr + i + 1), *(ptr + i + 2) , *(ptr + i + 3) , *(ptr + i + 4), *(ptr + i + 5), *(ptr + i + 6), *(ptr + i + 7));
+			}
 				RTW_INFO("#############################\n");
 				_rtw_memset(pmppriv->mplink_buf, '\0' , sizeof(pmppriv->mplink_buf));
 				_rtw_memcpy(pmppriv->mplink_buf, ptr, precv_frame->u.hdr.len);
diff -Naur a/drivers/net/wireless/rtl8812au/core/efuse/rtw_efuse.c b/drivers/net/wireless/rtl8812au/core/efuse/rtw_efuse.c
--- a/drivers/net/wireless/rtl8812au/core/efuse/rtw_efuse.c	2021-10-08 18:41:11.624166231 +0000
+++ b/drivers/net/wireless/rtl8812au/core/efuse/rtw_efuse.c	2021-10-08 21:21:28.422963823 +0000
@@ -907,8 +907,9 @@
 	j = 0;
 
 	for (i = 0; i < mapLen; i++) {
-		if (i % 16 == 0)
+		if (i % 16 == 0) {
 			RTW_PRINT_SEL(RTW_DBGDUMP, "0x%03x: ", i);
+		}
 			_RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
 				, pEfuseHal->fakeEfuseInitMap[i]
 				, ((i + 1) % 16 == 0) ? "\n" : (((i + 1) % 8 == 0) ? "	  " : " ")
diff -Naur a/drivers/net/wireless/rtl8812au/core/rtw_recv.c b/drivers/net/wireless/rtl8812au/core/rtw_recv.c
--- a/drivers/net/wireless/rtl8812au/core/rtw_recv.c	2021-10-08 18:41:11.708166044 +0000
+++ b/drivers/net/wireless/rtl8812au/core/rtw_recv.c	2021-10-08 21:11:42.340150682 +0000
@@ -3595,9 +3595,10 @@
 			int i;
 			RTW_INFO("############ type:0x%02x subtype:0x%02x #################\n", type, subtype);
 
-			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8)
+			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8) {
 				RTW_INFO("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X:\n", *(ptr + i),
 					*(ptr + i + 1), *(ptr + i + 2) , *(ptr + i + 3) , *(ptr + i + 4), *(ptr + i + 5), *(ptr + i + 6), *(ptr + i + 7));
+			}
 				RTW_INFO("#############################\n");
 				_rtw_memset(pmppriv->mplink_buf, '\0' , sizeof(pmppriv->mplink_buf));
 				_rtw_memcpy(pmppriv->mplink_buf, ptr, precv_frame->u.hdr.len);
diff -Naur a/drivers/net/wireless/rtl8814au/core/efuse/rtw_efuse.c b/drivers/net/wireless/rtl8814au/core/efuse/rtw_efuse.c
--- a/drivers/net/wireless/rtl8814au/core/efuse/rtw_efuse.c	2021-10-08 18:41:30.664123805 +0000
+++ b/drivers/net/wireless/rtl8814au/core/efuse/rtw_efuse.c	2021-10-08 21:19:04.959254528 +0000
@@ -902,8 +902,9 @@
 	j = 0;
 
 	for (i = 0; i < mapLen; i++) {
-		if (i % 16 == 0)
+		if (i % 16 == 0) {
 			RTW_PRINT_SEL(RTW_DBGDUMP, "0x%03x: ", i);
+		}
 			_RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
 				, pEfuseHal->fakeEfuseInitMap[i]
 				, ((i + 1) % 16 == 0) ? "\n" : (((i + 1) % 8 == 0) ? "	  " : " ")
diff -Naur a/drivers/net/wireless/rtl8814au/core/rtw_mlme.c b/drivers/net/wireless/rtl8814au/core/rtw_mlme.c
--- a/drivers/net/wireless/rtl8814au/core/rtw_mlme.c	2021-10-08 18:41:30.708123707 +0000
+++ b/drivers/net/wireless/rtl8814au/core/rtw_mlme.c	2021-10-08 20:47:04.642948109 +0000
@@ -3125,12 +3125,14 @@
 	ssc_chk = rtw_sitesurvey_condition_check(padapter, _FALSE);
 	if( ssc_chk == SS_DENY_BUSY_TRAFFIC) {
 		#ifdef CONFIG_LAYER2_ROAMING
-		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE)
+		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE) {
 			RTW_INFO(FUNC_ADPT_FMT" need to roam, don't care BusyTraffic\n", FUNC_ADPT_ARG(padapter));
-		else
+		} else
 		#endif
+		{
 			RTW_INFO(FUNC_ADPT_FMT" exit BusyTraffic\n", FUNC_ADPT_ARG(padapter));
 			goto exit;
+		}
 	}
 	else if (ssc_chk != SS_ALLOW)
 		goto exit;
diff -Naur a/drivers/net/wireless/rtl8814au/core/rtw_recv.c b/drivers/net/wireless/rtl8814au/core/rtw_recv.c
--- a/drivers/net/wireless/rtl8814au/core/rtw_recv.c	2021-10-08 18:41:30.736123645 +0000
+++ b/drivers/net/wireless/rtl8814au/core/rtw_recv.c	2021-10-08 21:03:51.037103091 +0000
@@ -3572,9 +3572,10 @@
 			int i;
 			RTW_INFO("############ type:0x%02x subtype:0x%02x #################\n", type, subtype);
 
-			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8)
+			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8) {
 				RTW_INFO("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X:\n", *(ptr + i),
 					*(ptr + i + 1), *(ptr + i + 2) , *(ptr + i + 3) , *(ptr + i + 4), *(ptr + i + 5), *(ptr + i + 6), *(ptr + i + 7));
+			}
 				RTW_INFO("#############################\n");
 				_rtw_memset(pmppriv->mplink_buf, '\0' , sizeof(pmppriv->mplink_buf));
 				_rtw_memcpy(pmppriv->mplink_buf, ptr, precv_frame->u.hdr.len);
diff -Naur a/drivers/net/wireless/rtl8822cs/core/efuse/rtw_efuse.c b/drivers/net/wireless/rtl8822cs/core/efuse/rtw_efuse.c
--- a/drivers/net/wireless/rtl8822cs/core/efuse/rtw_efuse.c	2021-10-08 18:41:26.992131984 +0000
+++ b/drivers/net/wireless/rtl8822cs/core/efuse/rtw_efuse.c	2021-10-08 21:30:16.625812359 +0000
@@ -778,8 +778,9 @@
 	j = 0;
 
 	for (i = 0; i < mapLen; i++) {
-		if (i % 16 == 0)
+		if (i % 16 == 0) {
 			RTW_PRINT_SEL(RTW_DBGDUMP, "0x%03x: ", i);
+		}
 			_RTW_PRINT_SEL(RTW_DBGDUMP, "%02X%s"
 				, pEfuseHal->fakeEfuseInitMap[i]
 				, ((i + 1) % 16 == 0) ? "\n" : (((i + 1) % 8 == 0) ? "	  " : " ")
diff -Naur a/drivers/net/wireless/rtl8822cs/core/rtw_mlme.c b/drivers/net/wireless/rtl8822cs/core/rtw_mlme.c
--- a/drivers/net/wireless/rtl8822cs/core/rtw_mlme.c	2021-10-08 18:41:27.040131877 +0000
+++ b/drivers/net/wireless/rtl8822cs/core/rtw_mlme.c	2021-10-08 20:47:18.642922824 +0000
@@ -3191,12 +3191,14 @@
 	ssc_chk = rtw_sitesurvey_condition_check(padapter, _FALSE);
 	if( ssc_chk == SS_DENY_BUSY_TRAFFIC) {
 		#ifdef CONFIG_LAYER2_ROAMING
-		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE)
+		if (rtw_chk_roam_flags(padapter, RTW_ROAM_ACTIVE) && pmlmepriv->need_to_roam == _TRUE) {
 			RTW_INFO(FUNC_ADPT_FMT" need to roam, don't care BusyTraffic\n", FUNC_ADPT_ARG(padapter));
-		else
+		} else
 		#endif
+		{
 			RTW_INFO(FUNC_ADPT_FMT" exit BusyTraffic\n", FUNC_ADPT_ARG(padapter));
 			goto exit;
+		}
 	}
 	else if (ssc_chk != SS_ALLOW)
 		goto exit;
diff -Naur a/drivers/net/wireless/rtl8822cs/core/rtw_recv.c b/drivers/net/wireless/rtl8822cs/core/rtw_recv.c
--- a/drivers/net/wireless/rtl8822cs/core/rtw_recv.c	2021-09-21 17:36:33.378916540 +0000
+++ b/drivers/net/wireless/rtl8822cs/core/rtw_recv.c	2021-10-08 21:06:53.364735113 +0000
@@ -3555,9 +3555,10 @@
 			int i;
 			RTW_INFO("############ type:0x%02x subtype:0x%02x #################\n", type, subtype);
 
-			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8)
+			for (i = 0; i < precv_frame->u.hdr.len; i = i + 8) {
 				RTW_INFO("%02X:%02X:%02X:%02X:%02X:%02X:%02X:%02X:\n", *(ptr + i),
 					*(ptr + i + 1), *(ptr + i + 2) , *(ptr + i + 3) , *(ptr + i + 4), *(ptr + i + 5), *(ptr + i + 6), *(ptr + i + 7));
+			}
 				RTW_INFO("#############################\n");
 				_rtw_memset(pmppriv->mplink_buf, '\0' , sizeof(pmppriv->mplink_buf));
 				_rtw_memcpy(pmppriv->mplink_buf, ptr, precv_frame->u.hdr.len);
--- a/drivers/net/wireless/rtl8822cs/os_dep/linux/ioctl_cfg80211.c	2022-02-13 19:04:07.946515034 +0200
+++ b/drivers/net/wireless/rtl8822cs/os_dep/linux/ioctl_cfg80211.c	2022-02-13 19:05:34.442821706 +0200
@@ -4483,6 +4483,7 @@
 }
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31))
+#ifdef CONFIG_RTW_DEBUG
 static const char *nl80211_tx_power_setting_str(int type)
 {
 	switch (type) {
@@ -4496,6 +4497,7 @@
 		return "UNKNOWN";
 	};
 }
+#endif	/*	CONFIG_RTW_DEBUG	*/
 
 static int cfg80211_rtw_set_txpower(struct wiphy *wiphy,
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
--- a/drivers/net/wireless/rtl8189es/os_dep/linux/ioctl_cfg80211.c	2022-05-25 16:24:47.194958238 +0300
+++ b/drivers/net/wireless/rtl8189es/os_dep/linux/ioctl_cfg80211.c	2022-05-25 16:25:03.127012179 +0300
@@ -4120,7 +4120,7 @@
 }
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 31))
-static const char *nl80211_tx_power_setting_str(int type)
+/*static const char *nl80211_tx_power_setting_str(int type)
 {
 	switch (type) {
 	case NL80211_TX_POWER_AUTOMATIC:
@@ -4132,7 +4132,7 @@
 	default:
 		return "UNKNOWN";
 	};
-}
+}*/
 
 static int cfg80211_rtw_set_txpower(struct wiphy *wiphy,
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
