--- a/drivers/net/wireless/rtl8189es/include/osdep_service_linux.h	2022-02-08 19:54:34.414685907 +0200
+++ b/drivers/net/wireless/rtl8189es/include/osdep_service_linux.h	2022-02-08 19:57:00.967123658 +0200
@@ -165,7 +165,11 @@
 	typedef int		thread_return;
 	typedef void*	thread_context;
 
-	#define thread_exit() complete_and_exit(NULL, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
+#define thread_exit() complete_and_exit(NULL, 0)
+#else
+#define thread_exit() kthread_complete_and_exit(NULL, 0)
+#endif
 
 	typedef void timer_hdl_return;
 	typedef void* timer_hdl_context;
