--- a/drivers/net/wireless/rtl8189es/include/osdep_service_linux.h	2022-02-08 19:54:34.414685907 +0200
+++ b/drivers/net/wireless/rtl8189es/include/osdep_service_linux.h	2022-02-08 19:57:00.967123658 +0200
@@ -165,7 +165,11 @@
 	typedef int		thread_return;
 	typedef void*	thread_context;
 
-	#define thread_exit() complete_and_exit(NULL, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 17, 0)
+#define thread_exit() complete_and_exit(NULL, 0)
+#else
+#define thread_exit() kthread_complete_and_exit(NULL, 0)
+#endif
 
 	typedef void timer_hdl_return;
 	typedef void* timer_hdl_context;
--- a/drivers/net/wireless/rtl8189es/os_dep/linux/rtw_proc.c	2022-02-08 20:06:33.628795920 +0200
+++ b/drivers/net/wireless/rtl8189es/os_dep/linux/rtw_proc.c	2022-02-08 20:08:51.009188638 +0200
@@ -44,6 +44,8 @@
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3,10,0))
 #define PDE_DATA(inode) PDE((inode))->data
 #define proc_get_parent_data(inode) PDE((inode))->parent->data
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 17, 0))
+#define PDE_DATA(inode) pde_data(inode)
 #endif
 
 #if(LINUX_VERSION_CODE < KERNEL_VERSION(2,6,24))
