--- a/drivers/net/wireless/Kconfig	2022-08-26 21:23:25.046339728 +0300
+++ b/drivers/net/wireless/Kconfig	2022-08-27 01:03:59.170410180 +0300
@@ -49,6 +50,7 @@ source "drivers/net/wireless/rtl8189fs/K
 source "drivers/net/wireless/rtl8189es/Kconfig"
 source "drivers/net/wireless/zydas/Kconfig"
 source "drivers/net/wireless/quantenna/Kconfig"
+source "drivers/net/wireless/xr829/Kconfig"
 
 config PCMCIA_RAYCS
 	tristate "Aviator/Raytheon 2.4GHz wireless support"
--- a/drivers/net/wireless/Makefile	2022-08-26 21:23:25.034339682 +0300
+++ a/drivers/net/wireless/Makefile	2022-08-27 01:18:46.141640030 +0300
@@ -44,3 +45,4 @@ obj-$(CONFIG_RTL8822CS) += rtl8822cs/
 obj-$(CONFIG_RTL8723DS) += rtl8723ds/
 obj-$(CONFIG_RTL8723DU) += rtl8723du/
 obj-$(CONFIG_RTL8814AU) += rtl8814au/
+obj-$(CONFIG_XR829_WLAN) += xr829/
--- a/drivers/net/wireless/xr829/Kconfig	2022-08-27 01:34:20.233245936 +0300
+++ b/drivers/net/wireless/xr829/Kconfig	2022-08-27 01:34:26.549271048 +0300
@@ -5,5 +5,5 @@ config XR829_WLAN
 	select AVERAGE
 	select CRYPTO
 	select CRYPTO_ARC4
-	---help---
+	help
 	  To compile this code as a module, choose M here.
