From f0120ddb2632d2e828ffa2920b0163e2a2240e8d Mon Sep 17 00:00:00 2001
From: mara <mara@fail.pp.ua>
Date: Sat, 10 Jun 2017 15:25:49 +0300
Subject: update kernel source


diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-core.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-firefly-core.dtsi
index 7a3e09ed..d5b50e3c 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-core.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-core.dtsi
@@ -246,7 +246,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&rgmii_pins>;
 	tx_delay = <0x28>;
-	rx_delay = <0x11>;
+	rx_delay = <0x1B>;
 	status = "okay";
 };
 
