--- a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2018-11-25 16:59:50.690681295 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2018-11-25 16:57:40.977008207 +0200
@@ -1046,6 +1046,19 @@
 	extcon = <&fusb0>;
 };
 
+
+&vpu {
+        status = "okay";
+        /* 0 means ion, 1 means drm */
+        //allocator = <0>;
+};
+
+&rkvdec {
+        status = "okay";
+        /* 0 means ion, 1 means drm */
+        //allocator = <0>;
+};
+
 &display_subsystem {
 	/delete-property/ devfreq;
 	status = "okay";
