--- a/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2018-11-27 21:33:23.431899971 +0200
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rockpro64.dts	2018-11-27 21:32:31.370204542 +0200
@@ -1046,6 +1046,19 @@
 	extcon = <&fusb0>;
 };
 
+
+&vpu {
+        status = "okay";
+        /* 0 means ion, 1 means drm */
+        //allocator = <0>;
+};
+
+&rkvdec {
+        status = "okay";
+        /* 0 means ion, 1 means drm */
+        //allocator = <0>;
+};
+
 &display_subsystem {
 	/delete-property/ devfreq;
 	status = "okay";
@@ -1161,10 +1174,10 @@
 	i2c-scl-rising-time-ns = <168>;
 	i2c-scl-falling-time-ns = <4>;
 
-	es8316: es8316@10 {
+	es8316: es8316@11 {
 		#sound-dai-cells = <0>;
 		compatible = "everest,es8316";
-		reg = <0x10>;
+		reg = <0x11>;
 		clocks = <&cru SCLK_I2S_8CH_OUT>;
 		clock-names = "mclk";
 		pinctrl-names = "default";
