--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi	2019-08-13 20:08:54.365380817 +0300
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi	2019-09-08 13:08:29.892118819 +0300
@@ -793,7 +743,7 @@
 			polling-delay = <1000>;
 			sustainable-power = <300>;
 
-			thermal-sensors = <&tsadc 0>;
+			thermal-sensors = <&tsadc 1>;
 
 			trips {
 				threshold: trip-point-0 {
@@ -827,7 +777,7 @@
 			polling-delay-passive = <100>; /* milliseconds */
 			polling-delay = <1000>; /* milliseconds */
 
-			thermal-sensors = <&tsadc 1>;
+			thermal-sensors = <&tsadc 0>;
 		};
 	};
 
