--- a/drivers/mfd/fusb302.c	2018-02-10 17:33:26.526514823 +0200
+++ b/drivers/mfd/fusb302.c	2018-02-11 03:41:07.713846073 +0200
@@ -1161,7 +1161,7 @@
 	switch (chip->vdm_send_state) {
 	case 0:
 		set_vdm_mesg(chip, VDM_DISCOVERY_SVIDS, VDM_TYPE_INIT, 0);
-		memset(chip->vdm_svid, 0, 12);
+		memset(chip->vdm_svid, 0, sizeof(chip->vdm_svid));
 		chip->vdm_svid_num = 0;
 		chip->tx_state = 0;
 		chip->vdm_send_state++;
