diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index 94cf5a1..70b4a46 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
--- b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -107,6 +107,11 @@
 			clock-latency = <244144>; /* 8 32k periods */
 			operating-points = <
 				/* kHz	  uV */
+			       1200000	1500000
+			       1152000	1500000
+			       1104000	1450000
+			       1056000	1450000
+			       1008000	1450000
 				960000	1400000
 				912000	1400000
 				864000	1300000
diff --git a/arch/arm/boot/dts/sun7i-a20-cubietruck.dts b/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
index e43267c..6ed1719 100644
--- a/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
+++ b/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
@@ -285,14 +285,14 @@
 &reg_dcdc2 {
 	regulator-always-on;
 	regulator-min-microvolt = <1000000>;
-	regulator-max-microvolt = <1450000>;
+	regulator-max-microvolt = <1500000>;
 	regulator-name = "vdd-cpu";
 };
 
 &reg_dcdc3 {
 	regulator-always-on;
 	regulator-min-microvolt = <1000000>;
-	regulator-max-microvolt = <1400000>;
+	regulator-max-microvolt = <1450000>;
 	regulator-name = "vdd-int-dll";
 };
 
