--- a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts	2019-12-06 21:03:00.310589158 +0200
+++ a/arch/arm64/boot/dts/rockchip/rk3308-rock-pi-s.dts	2019-12-06 21:04:17.710261935 +0200
@@ -370,7 +370,7 @@
 };
 
 &usb20_otg {
-	dr_mode = "host";
+	dr_mode = "otg";
 	status = "okay";
 };
 
@@ -385,3 +385,9 @@
 &acodec {
 	status = "okay";
 };
+
+&thermal_zones {
+	soc_thermal {
+		thermal-sensors = <&tsadc 0>;
+	};
+};
