From 7da35aca65955179cd9162de5d60b558b1551d14 Mon Sep 17 00:00:00 2001
From: Jack Ma <jack@radxa.com>
Date: Sat, 17 Jul 2021 15:03:56 +0800
Subject: [PATCH] Use distro_boot by default

Signed-off-by: Jack Ma <jack@radxa.com>
---
 include/configs/radxa-zero.h | 26 +++++++++++++++++++++++---
 1 file changed, 23 insertions(+), 3 deletions(-)

diff --git a/include/configs/radxa-zero.h b/include/configs/radxa-zero.h
index d69922131f3..a1ad23c4f74 100644
--- a/include/configs/radxa-zero.h
+++ b/include/configs/radxa-zero.h
@@ -9,13 +9,33 @@
 #ifndef __CONFIG_H
 #define __CONFIG_H
 
-#define LOGO_UUID "43a3305d-150f-4cc9-bd3b-38fca8693846;"
-#define ROOT_UUID "ddb8c3f6-d94d-4394-b633-3134139cc2e0;"
+#define BOOT_UUID "7E0DC62E-464E-48FE-B1E0-6457A12F3511;"
+#define ROOT_UUID "6604DF74-56FE-490C-BC88-AF86667B84FF;"
 
 #define PARTS_DEFAULT                                        \
 	"uuid_disk=${uuid_gpt_disk};"  	\
+	"name=boot,size=512M,bootable,uuid=" BOOT_UUID \
 	"name=rootfs,size=-,uuid=" ROOT_UUID
 
-#include <configs/meson64_android.h>
+#define PREBOOT_LOAD_LOGO \
+		"if fatload ${mmcdev}:1 ${loadaddr} logo.bmp; then " \
+			"bmp display ${loadaddr} m m;" \
+		"fi;" \
+
+#define CONFIG_EXTRA_ENV_SETTINGS \
+	"stdin=" STDIN_CFG "\0" \
+	"stdout=" STDOUT_CFG "\0" \
+	"stderr=" STDOUT_CFG "\0" \
+	"partitions=" PARTS_DEFAULT "\0" \
+	"load_logo=" PREBOOT_LOAD_LOGO "\0" \
+	"fdt_addr_r=0x08008000\0" \
+	"scriptaddr=0x08000000\0" \
+	"kernel_addr_r=0x08080000\0" \
+	"pxefile_addr_r=0x01080000\0" \
+	"ramdisk_addr_r=0x13000000\0" \
+	"fdtfile=amlogic/" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
+	BOOTENV
+
+#include <configs/meson64.h>
 
 #endif /* __CONFIG_H */
