--- a/arch/arm64/boot/dts/amlogic/Makefile	2022-04-12 18:37:24.289921620 +0300
+++ a/arch/arm64/boot/dts/amlogic/Makefile	2022-04-12 18:37:35.497963254 +0300
@@ -18,5 +18,5 @@
 dtb-$(CONFIG_ARCH_MESON64_ODROIDN2) += meson64_odroidn2_drm.dtb
 dtb-$(CONFIG_ARCH_MESON64_ODROIDN2) += meson-g12b-odroid-n2-drm.dtb
 
-subdir-$(CONFIG_ARCH_MESON64_ODROIDC4) += overlays/odroidc4
-subdir-$(CONFIG_ARCH_MESON64_ODROIDN2) += overlays/odroidn2
+subdir-$(CONFIG_ARCH_MESON64_ODROIDC4) += overlay/odroidc4
+subdir-$(CONFIG_ARCH_MESON64_ODROIDN2) += overlay/odroidn2
