--- a/arch/arm/boot/dts/rk3288-firefly.dtsi	2019-08-31 01:30:59.029214879 +0300
+++ b/arch/arm/boot/dts/rk3288-firefly.dtsi	2019-08-31 01:33:34.019235815 +0300
@@ -213,8 +213,8 @@
 		fcs,suspend-voltage-selector = <1>;
 		reg = <0x40>;
 		regulator-name = "vdd_cpu";
-		regulator-min-microvolt = <850000>;
-		regulator-max-microvolt = <1350000>;
+		regulator-min-microvolt = <712500>;
+		regulator-max-microvolt = <1500000>;
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-enable-ramp-delay = <300>;
@@ -227,8 +227,8 @@
 		fcs,suspend-voltage-selector = <1>;
 		reg = <0x41>;
 		regulator-name = "vdd_gpu";
-		regulator-min-microvolt = <850000>;
-		regulator-max-microvolt = <1350000>;
+		regulator-min-microvolt = <712500>;
+		regulator-max-microvolt = <1500000>;
 		regulator-always-on;
 		vin-supply = <&vcc_sys>;
 	};
