diff -Naur rkwifi_/bcmdhd/Kconfig rkwifi/bcmdhd/Kconfig
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig	2017-02-01 18:14:09.509115251 +0200
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig	2017-02-01 20:52:46.314912852 +0200
@@ -7,14 +7,14 @@
 config BCMDHD_FW_PATH
 	depends on BCMDHD
 	string "Firmware path"
-	default "/system/etc/firmware/fw_bcmdhd.bin"
+	default "/lib/firmware/fw_bcmdhd.bin"
 	---help---
 	  Path to the firmware file.
 
 config BCMDHD_NVRAM_PATH
 	depends on BCMDHD
 	string "NVRAM path"
-	default "/system/etc/firmware/nvram.txt"
+	default "/lib/firmware/nvram.txt"
 	---help---
 	  Path to the calibration file.
 
diff -Naur rkwifi_/rk_wifi_config.c rkwifi/rk_wifi_config.c
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c	2017-02-01 18:14:09.562115249 +0200
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c	2017-02-01 20:52:48.077912773 +0200
@@ -16,7 +16,7 @@
  * Set Firmware Path
  */
  
-#define ANDROID_FW_PATH "/system/etc/firmware/"
+#define ANDROID_FW_PATH "/lib/firmware/"
 
 extern int get_wifi_chip_type(void);
 int rkwifi_set_firmware(char *fw, char *nvram)
