--- a/arch/arm/boot/dts/rk3288.dtsi	2017-02-05 15:18:10.625731329 +0200
+++ b/arch/arm/boot/dts/rk3288.dtsi	2017-02-05 15:21:29.999701294 +0200
@@ -174,14 +174,24 @@
 		};
 		opp@1512000000 {
 			opp-hz = /bits/ 64 <1512000000>;
-			opp-microvolt = <1300000>;
+			opp-microvolt = <1250000>;
 			clock-latency-ns = <40000>;
 		};
 		opp@1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1300000>;
+			clock-latency-ns = <40000>;
+		};
+		opp@1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
 			opp-microvolt = <1350000>;
 			clock-latency-ns = <40000>;
 		};
+		opp@1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1400000>;
+			clock-latency-ns = <40000>;
+		};
 	};
 
 	cpu_avs: cpu-avs {
--- a/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-01-18 18:38:00.735038093 +0200
+++ b/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-02-05 16:09:18.040230639 +0200
@@ -295,7 +295,7 @@
 		reg = <0x40>;
 		regulator-name = "vdd_cpu";
 		regulator-min-microvolt = <850000>;
-		regulator-max-microvolt = <1350000>;
+		regulator-max-microvolt = <1400000>;
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-enable-ramp-delay = <300>;
@@ -309,7 +309,7 @@
 		reg = <0x41>;
 		regulator-name = "vdd_gpu";
 		regulator-min-microvolt = <850000>;
-		regulator-max-microvolt = <1350000>;
+		regulator-max-microvolt = <1400000>;
 		regulator-always-on;
 		regulator-ramp-delay = <6000>;
 		vin-supply = <&vcc_sys>;
