--- a/drivers/net/wireless/realtek/Kconfig	2021-01-17 21:12:12.284353838 +0200
+++ b/drivers/net/wireless/realtek/Kconfig	2021-01-17 21:05:07.281782866 +0200
@@ -16,5 +16,6 @@
 source "drivers/net/wireless/realtek/rtlwifi/Kconfig"
 source "drivers/net/wireless/realtek/rtl8xxxu/Kconfig"
 source "drivers/net/wireless/realtek/rtw88/Kconfig"
+source "drivers/net/wireless/realtek/rtl8822bu/Kconfig"
 
 endif # WLAN_VENDOR_REALTEK
--- a/drivers/net/wireless/realtek/Makefile	2021-01-17 21:12:23.536317672 +0200
+++ b/drivers/net/wireless/realtek/Makefile	2021-01-17 21:05:31.441698229 +0200
@@ -8,4 +8,5 @@
 obj-$(CONFIG_RTLWIFI)		+= rtlwifi/
 obj-$(CONFIG_RTL8XXXU)		+= rtl8xxxu/
 obj-$(CONFIG_RTW88)		+= rtw88/
+obj-$(CONFIG_RTL8822BU)		+= rtl8822bu/
 
