diff -Naur a/arch/arm/boot/dts/rk3288-firefly.dts b/arch/arm/boot/dts/rk3288-firefly.dts
--- a/arch/arm/boot/dts/rk3288-firefly.dts	2017-12-29 22:44:21.147736859 +0200
+++ b/arch/arm/boot/dts/rk3288-firefly.dts	2017-12-29 22:45:33.233739623 +0200
@@ -67,5 +67,6 @@
 };
 
 &pwm1 {
-	status = "okay";
+	status = "disabled";
+	//status = "okay";
 };
diff -Naur a/arch/arm/boot/dts/rk3288-firefly.dtsi b/arch/arm/boot/dts/rk3288-firefly.dtsi
--- a/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-12-29 22:44:21.147736859 +0200
+++ b/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-12-29 22:45:33.022739614 +0200
@@ -49,6 +49,7 @@
 	};
 
 	backlight: backlight {
+		status = "disabled";
 		compatible = "pwm-backlight";
 		brightness-levels = <
 			  3   4   5   6   7
@@ -89,7 +90,8 @@
 		power-supply = <&vcc_sys>;
 	};
 
-	edp_panel: edp-panel {
+	panel: panel {
+		status = "disabled";
 		compatible ="lg,lp079qx1-sp0v";
 		power-supply = <&vcc_sys>;
 		backlight = <&backlight>;
@@ -176,6 +178,29 @@
 		reset-gpios = <&gpio4 28 GPIO_ACTIVE_LOW>;
 	};
 
+	es8323-sound {
+		status = "okay";
+        compatible = "simple-audio-card";
+        simple-audio-card,format = "i2s";
+        simple-audio-card,name = "rockchip,es8323-codec";
+        simple-audio-card,mclk-fs = <256>;
+        //simple-audio-card,hp-det-gpio = <&gpio7 15 GPIO_ACTIVE_HIGH>;
+		simple-audio-card,widgets =
+				"Microphone","Mic Jack",
+				"Headphone","Headphone Jack";
+		simple-audio-card,routing =
+				"Mic Jack","MICBIAS1",
+				"IN1P","Mic Jack",
+				"Headphone Jack","HPOL",
+				"Headphone Jack","HPOR";
+		simple-audio-card,cpu {
+				sound-dai = <&i2s>;
+		};
+		simple-audio-card,codec {
+				sound-dai = <&es8323>;
+		};
+	};
+
 	sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "SPDIF";
@@ -302,7 +327,7 @@
 	vcc-supply = <&vdd_10>;
 	vccio-supply = <&vcc_18>;
 	force-hpd;
-	status = "okay";
+	status = "disabled";
 
 	ports {
 		edp_out: port@1 {
@@ -317,8 +342,13 @@
 	};
 };
 
+&edp_in {
+	/delete-node/ endpoint@0;
+};
+
 &edp_phy {
-	status = "okay";
+	status = "disabled";
+	//status = "okay";
 };
 
 &emmc {
@@ -364,6 +394,10 @@
 
 };
 
+&hdmi_in {
+	/delete-node/ endpoint@1;
+};
+
 &i2c0 {
 	clock-frequency = <400000>;
 	status = "okay";
@@ -529,6 +563,17 @@
 
 &i2c2 {
 	status = "okay";
+
+	es8323: es8323@10 {
+        status = "okay";
+		compatible = "everest,es8323";
+		reg = <0x10>;
+		spk-con-gpio = <&gpio7 3 GPIO_ACTIVE_HIGH>;
+		hp-det-gpio = <&gpio7 15 GPIO_ACTIVE_LOW>;
+		clock-names = "mclk";
+		clocks = <&cru SCLK_I2S0_OUT>;
+		#sound-dai-cells = <0>;
+	};
 };
 
 &i2c4 {
@@ -674,6 +719,11 @@
 	};
 };
 
+&i2s {
+	status = "okay";
+	#sound-dai-cells = <0>;
+};
+
 &saradc {
 	vref-supply = <&vcc_18>;
 	status = "okay";
@@ -771,6 +821,10 @@
 	status = "okay";
 };
 
+&vopb_out {
+	/delete-node/ endpoint@1;
+};
+
 &vopl {
 	status = "okay";
 };
@@ -783,6 +837,10 @@
 	status = "okay";
 };
 
+&vopl_out {
+	/delete-node/ endpoint@0;
+};
+
 &wdt {
 	status = "okay";
 };
