--- a/arch/arm/boot/dts/rk3288.dtsi     2018-01-18 21:02:03.280447007 +0200
+++ b/arch/arm/boot/dts/rk3288.dtsi     2018-01-18 21:09:39.891426534 +0200
@@ -150,6 +150,7 @@
 		rockchip,pvtm-thermal-zone = "soc-thermal";
 
 		opp-126000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <126000000>;
 			opp-microvolt = <900000>;
 			opp-microvolt-L0 = <900000>;
@@ -158,6 +159,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-216000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <216000000>;
 			opp-microvolt = <900000>;
 			opp-microvolt-L0 = <900000>;
@@ -166,6 +168,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-408000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <900000>;
 			opp-microvolt-L0 = <900000>;
@@ -174,6 +177,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <900000>;
 			opp-microvolt-L0 = <900000>;
@@ -182,6 +186,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-696000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <696000000>;
 			opp-microvolt = <950000>;
 			opp-microvolt-L0 = <950000>;
@@ -190,6 +195,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <1050000>;
 			opp-microvolt-L0 = <1050000>;
@@ -199,6 +205,7 @@
 			opp-suspend;
 		};
 		opp-1008000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <1100000>;
 			opp-microvolt-L0 = <1100000>;
@@ -207,6 +214,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <1150000>;
 			opp-microvolt-L0 = <1150000>;
@@ -215,6 +223,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-1416000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1250000>;
 			opp-microvolt-L0 = <1250000>;
@@ -223,6 +232,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-1512000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1512000000>;
 			opp-microvolt = <1300000>;
 			opp-microvolt-L0 = <1300000>;
@@ -231,6 +241,7 @@
 			clock-latency-ns = <40000>;
 		};
 		opp-1608000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1608000000>;
 			opp-microvolt = <1350000>;
 			opp-microvolt-L0 = <1350000>;
@@ -238,6 +249,24 @@
 			opp-microvolt-L2 = <1250000>;
 			clock-latency-ns = <40000>;
 		};
+		opp-1704000000 {
+			opp-supported-hw = <0x3>;
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1375000>;
+			opp-microvolt-L0 = <1375000>;
+			opp-microvolt-L1 = <1325000>;
+			opp-microvolt-L2 = <1275000>;
+			clock-latency-ns = <40000>;
+		};
+		opp-1800000000 {
+			opp-supported-hw = <0x3>;
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1400000>;
+			opp-microvolt-L0 = <1400000>;
+			opp-microvolt-L1 = <1350000>;
+			opp-microvolt-L2 = <1300000>;
+			clock-latency-ns = <40000>;
+		};
 	};
 
 	amba {
