--- a/arch/arm/boot/dts/rk3288.dtsi     2018-01-06 19:02:34.426350402 +0200
+++ b/arch/arm/boot/dts/rk3288.dtsi     2018-01-06 19:08:17.616363557 +0200
@@ -134,62 +134,88 @@
 		nvmem-cells = <&cpu_leakage>;
 		nvmem-cell-names = "cpu_leakage";
 
-		opp-126000000 {
+		opp@126000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <126000000>;
 			opp-microvolt = <900000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-216000000 {
+		opp@216000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <216000000>;
 			opp-microvolt = <900000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-408000000 {
+		opp@408000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <408000000>;
 			opp-microvolt = <900000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-600000000 {
+		opp@600000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <600000000>;
 			opp-microvolt = <900000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-696000000 {
+		opp@696000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <696000000>;
 			opp-microvolt = <950000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-816000000 {
+		opp@816000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <816000000>;
 			opp-microvolt = <1000000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
-		opp-1008000000 {
+		opp@1008000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1008000000>;
 			opp-microvolt = <1050000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1200000000 {
+		opp@1200000000 {	
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <1100000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1416000000 {
+		opp@1416000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1416000000>;
 			opp-microvolt = <1200000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1512000000 {
+		opp@1512000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1512000000>;
 			opp-microvolt = <1300000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1608000000 {
+
+		opp@1608000000 {
+			opp-supported-hw = <0x3>;
 			opp-hz = /bits/ 64 <1608000000>;
 			opp-microvolt = <1350000>;
 			clock-latency-ns = <40000>;
 		};
+
+		opp@1704000000 {
+			opp-supported-hw = <0x3>;
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1350000>;
+			clock-latency-ns = <40000>;
+		};
+
+		opp@1800000000 {
+			opp-supported-hw = <0x3>;
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1350000>;
+			clock-latency-ns = <40000>;
+		};
 	};
 
 	amba {
